packages/core/tests/utils/AssetUtil.test.ts

```diff
@@ -0,0 +1,99 @@
+import { beforeEach, describe, expect, it, vi } from 'vitest'
+import { AssetController } from '../../src/controllers/AssetController.js'
+import { AssetUtil } from '../../src/utils/AssetUtil.js'
+import type { CaipNetwork, Connector, WcWallet } from '../../src/utils/TypeUtil.js'
+import { ApiController } from '../../src/controllers/ApiController.js'
+
+// - Mocks ---------------------------------------------------------------------
+const connector: Partial<Connector> = {
+  imageUrl: undefined,
+  imageId: 'walletconnect'
+}
+const connectorWithImageUrl: Partial<Connector> = {
+  imageUrl: 'walletconnect-connector-logo-src',
+  imageId: 'walletconnect'
+}
+const network: Partial<CaipNetwork> = {
+  imageUrl: undefined,
+  imageId: 'ethereum'
+}
+const networkWithImageUrl: Partial<Connector> = {
+  imageUrl: 'ethereum-logo-src',
+  imageId: 'ethereum'
+}
+const wallet: Partial<WcWallet> = {
+  image_url: undefined,
+  image_id: 'metamask'
+}
+const walletWithImageUrl: Partial<WcWallet> = {
+  image_url: 'metamask-logo-src',
+  image_id: 'metamask'
+}
+
+// -- Setup --------------------------------------------------------------------
+ApiController._fetchWalletImage = vi.fn().mockImplementation((walletId: string) => {
+  AssetController.state = {
+    ...AssetController.state,
+    walletImages: {
+      ...AssetController.state.walletImages,
+      [walletId]: `new-wallet-image-url-${walletId}`
+    }
+  }
+})
+beforeEach(() => {
+  AssetController.state = {
+    ...AssetController.state,
+    connectorImages: {
+      walletconnect: 'walletconnect-connector-logo-blob-url'
+    },
+    walletImages: {
+      metamask: 'metamask-logo-blob-url'
+    },
+    networkImages: {
+      ethereum: 'ethereum-logo-blob-url'
+    }
+  }
+})
+
+// -- Tests --------------------------------------------------------------------
+describe('AssetUtil', () => {
+  it('should get the connector image from connector object', () => {
+    // @ts-expect-error it's a partial connector object
```

Comment:
I'd apply the disable globally instead of on every test

Reply:
Good idea, I don't have strong opinion but I wouldn't want to discard all benefits of the TS, instead stoping type checking (with `@ts-nocheck` on top of the file), only use disable TS for a single thing sounds a better IMO

Reply:
given it's a partial you could maybe cast it to `Partial<Connector>` ?
Else agree with your point but spamming disables is kind of the same thing 🤣